src/services/etherjs/provider.ts

```diff
@@ -2,15 +2,15 @@ import { ethers, providers } from "ethers";
 import { getLogger } from "../../utils/logger";
 const { trace } = getLogger("service:etherjs:provider");
 
-interface Ethereum extends providers.Web3Provider {
+interface Ethereum extends providers.ExternalProvider, providers.BaseProvider {
```

Comment:
Web3Provider expected param of type `ExternalProvider` but windows.ethereum.on method was part of the BaseProvider class, not sure if this is right!

Reply:
Previously, type  `provider.Web3Provider` looked more like the `windows.web3` structure, [refer here for more info](https://github.com/ethers-io/ethers.js/blob/master/packages/providers/lib/web3-provider.d.ts).